addons/binding/org.openhab.binding.ftpupload/src/main/java/org/openhab/binding/ftpupload/internal/ftp/FtpServer.java

```diff
@@ -0,0 +1,207 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.ftpupload.internal.ftp;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.ftpserver.FtpServerConfigurationException;
+import org.apache.ftpserver.FtpServerFactory;
+import org.apache.ftpserver.ftplet.DefaultFtplet;
+import org.apache.ftpserver.ftplet.FileSystemFactory;
+import org.apache.ftpserver.ftplet.FileSystemView;
+import org.apache.ftpserver.ftplet.FtpException;
+import org.apache.ftpserver.ftplet.FtpRequest;
+import org.apache.ftpserver.ftplet.FtpSession;
+import org.apache.ftpserver.ftplet.FtpStatistics;
+import org.apache.ftpserver.ftplet.Ftplet;
+import org.apache.ftpserver.ftplet.FtpletContext;
+import org.apache.ftpserver.ftplet.FtpletResult;
+import org.apache.ftpserver.ftplet.User;
+import org.apache.ftpserver.listener.Listener;
+import org.apache.ftpserver.listener.ListenerFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Simple FTP server implementation to receive files via FTP.
+ *
+ *
+ * @author Pauli Anttila - Initial contribution
+ */
+public class FtpServer {
+
+    private Logger logger = LoggerFactory.getLogger(FtpServer.class);
+
+    private int port;
+    int idleTimeout;
+
+    private org.apache.ftpserver.FtpServer server;
+    private static List<FtpServerEventListener> listeners;
+    private MyFTPLet myFTPLet;
+    private FTPUserManager FTPUserManager;
+    private String ftpStartUpErrorReason;
+
+    public FtpServer() {
+        listeners = new ArrayList<FtpServerEventListener>();
+        FTPUserManager = new FTPUserManager();
+    }
+
+    public void startServer(int port, int idleTimeout) throws FtpException {
+        stopServer();
+        this.port = port;
+        this.idleTimeout = idleTimeout;
+        FTPUserManager.setIdleTimeout(idleTimeout);
+        initServer();
+    }
+
+    public void stopServer() {
+        if (server != null) {
+            logger.info("Stopping FTP server");
```

Comment:
Such internal classes should better not do info logging - would be nicer to move it to the factory service, which instantiates the FTP server.

Reply:
Done